config/config.go

```diff
@@ -168,6 +171,30 @@ func SingleNetworkLookup(rpc string) map[string]*NodeConfig {
 	}
 }
 
+type CacheList struct {
+	// Consensus holds the configuration for the consensus cache
+	Consensus *CacheConfig `koanf:"consensus"`
+
+	// Runtime holds the configuration for the runtime cache
+	Runtime *CacheConfig `koanf:"runtime"`
+}
+
+type CacheConfig struct {
+	// File is the directory where the cache data is stored
+	File string `koanf:"file"`
+
+	// If set, the indexer will query the node upon any cache
+	// misses.
+	QueryOnCacheMiss bool `koanf:"query_on_cache_miss"`
```

Comment:
Is it possible to get rid of this flag? 
I see two scenarios:
- We expect cache misses. Then we want this to be true, obviously.
- We do not expect cache misses. Expected use case are offline e2e tests. If there really are no cache misses, great, and this flag doesn't matter. If there _is_ a cache miss though ... this flag will only change the behavior from getting a network error to getting some slightly different error message. That seems not worth it.

(It does become a more meaningful flag if we want to verify that a run has no cache misses even when we do have access to a node. But I don't see a use case for that, other than e2e tests, and CI will alert us to any problems there, albeit slightly less ergonomically. Or, to test before CI, one could just set the node URL to a bad string to ensure the analyzer is "offline".)

Reply:
Ah good eye; I don't think it's possible to remove the flag entirely but first some context:

This flag was originally intended to tell the analyzer whether to try to connect to a node on startup. This is necessary for two reasons. 1) During startup, the analyzer tries to connect to the node and if the connection cannot be made (either a bad url config or unreachable node), the analyzer errors out. 2) In CI(outside of docker-compose), a node connection will not be available. Therefore, the indexer needs to know whether to _attempt_ to connect to the node, and that was the original purpose of the `QueryOnCacheMiss` flag. 

However, during a later commit the logic was removed and the indexer currently always tries to connect to the node on startup. I think it's still important to allow file-based indexers to be able to run without a node connection (for CI / full reindexes) though, so I'm planning to re-add this logic in this PR, at least for the consensus analyzer. The runtime analyzer is a bit trickier to handle because it fetches the `runtimeInfo` from node during startup, so we'd need to store that offline in the config as well.

Reply:
Yes, it's crucial we allow fully-offline execution, CI being the strongest / most immediate reason.

Bummer about the underlying "real" ApiLite impls trying to connect immediately; I missed that. In an ideal world I would prefer to change those ApiLites so they connect lazily (= when the first RPC requires a connection) and then we wouldn't need this flag and its complications. But let's not go there; this flag is a fine alternative. Thanks for explaining!